File: ANSv4/Catalogo3/forms/frmErrorGuardian.vb
Imports System.Windows.Forms

Public Class frmErrorGuardian

    Private mModuloName As String
    Private mErrCompleto As String

    Public Property ModuloName() As String
        Get
            Return mModuloName
        End Get
        Set(ByVal value As String)
            mModuloName = value
        End Set
    End Property

    Public Property ErrCompleto() As String
        Get
            Return mErrCompleto
        End Get
        Set(ByVal value As String)
            mErrCompleto = value
        End Set
    End Property

    Public WriteOnly Property ErrDetails() As String
        Set(ByVal value As String)
            TextBox1.Text = value
            TextBox2.Text = vbNullString
        End Set
    End Property

    Private Sub btnAbortar_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnAbortar.Click

        Dim sRep As String

        ErrorGuardianUserReply = DialogResult.Abort
        sRep = "Abortar Pulsado..."

        'On Error Resume Next
        'vg.Auditor.Guardar ErrordePrograma, EXITOSO, sRep
        'On Error GoTo 0

        ErrLog(sRep)

        Me.DialogResult = System.Windows.Forms.DialogResult.Abort
        Me.Close()

    End Sub

    Private Sub btnIgnorar_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnIgnorar.Click

        Dim sRep As String

        ErrorGuardianUserReply = System.Windows.Forms.DialogResult.Ignore
        sRep = "Ignorar Pulsado..."

        'On Error Resume Next
        'vg.Auditor.Guardar ErrordePrograma, EXITOSO, sRep
        'On Error GoTo 0

        ErrLog(sRep)

        Me.DialogResult = System.Windows.Forms.DialogResult.Ignore
        Me.Close()
    End Sub

    Private Sub btnReintentar_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnReintentar.Click

        Dim sRep As String

        ErrorGuardianUserReply = System.Windows.Forms.DialogResult.Retry
        sRep = "Reintentar Pulsado..."

        'On Error Resume Next
        'vg.Auditor.Guardar ErrordePrograma, EXITOSO, sRep
        'On Error GoTo 0

        ErrLog(sRep)

        Me.DialogResult = System.Windows.Forms.DialogResult.Retry
        Me.Close()
    End Sub

    Private Sub ErrLog(ByVal sRep As String)

        Dim fn As String
        Static Dim Separator As String

        Separator = Strings.Space(78).Replace(" ", "-")

        fn = vg.Path & "\ErrorGuardianLog.txt"
        Dim file As System.IO.TextWriter = New System.IO.StreamWriter(fn)

        file.WriteLine(mErrCompleto)
        file.WriteLine("")
        If TextBox2.Text.Trim.Length > 0 Then
            file.WriteLine(TextBox2.Text)
            file.WriteLine("")
        End If
        'Print #FF, sRep
        file.WriteLine(Separator)
        file.Flush()
        file.Close()

    End Sub

End Class

File: ANSv4/Catalogo4/forms/frmErrorGuardian.cs
using System;
using System.Windows.Forms;

namespace Catalogo.Funciones
{
    public partial class frmErrorGuardian : Form
    {
        private System.Windows.Forms.DialogResult ErrorGuardianUserReply;

        private string BuildNumber;

        private string mModuloName;

        private string mErrCompleto;
        public string ModuloName
        {
            get { return mModuloName; }
            set { mModuloName = value; }
        }

        public string ErrCompleto
        {
            get { return mErrCompleto; }
            set { mErrCompleto = value; }
        }

        public string ErrDetails
        {
            set
            {
                TextBox1.Text = value;
                TextBox2.Text = null;
            }
        }


        private void btnAbortar_Click(System.Object sender, System.EventArgs e)
        {

        }


        private void btnIgnorar_Click(System.Object sender, System.EventArgs e)
        {
            string sRep = null;

            ErrorGuardianUserReply = System.Windows.Forms.DialogResult.Ignore;
            sRep = "Ignorar Pulsado...";

            //On Error Resume Next
            auditoria.Auditor.instance.guardar(auditoria.Auditor.ObjetosAuditados.ErrordePrograma,
                 auditoria.Auditor.AccionesAuditadas.EXITOSO, sRep);
            //On Error GoTo 0

            ErrLog(sRep);

            this.DialogResult = System.Windows.Forms.DialogResult.Ignore;
            this.Close();
        }


        private void btnReintentar_Click(System.Object sender, System.EventArgs e)
        {
            string sRep = null;

            ErrorGuardianUserReply = System.Windows.Forms.DialogResult.Retry;
            sRep = "Reintentar Pulsado...";

            //On Error Resume Next
            auditoria.Auditor.instance.guardar(auditoria.Auditor.ObjetosAuditados.ErrordePrograma,
                 auditoria.Auditor.AccionesAuditadas.EXITOSO, sRep);
            //On Error GoTo 0

            ErrLog(sRep);

            this.DialogResult = System.Windows.Forms.DialogResult.Retry;
            this.Close();
        }

        private void ErrLog(string sRep)
        {
            string static_ErrLog_Separator;

            string fn = null;

            static_ErrLog_Separator = new String(' ', 78).Replace(" ", "-");

            fn = Application.StartupPath + "\\ErrorGuardianLog.txt";
            System.IO.TextWriter file = new System.IO.StreamWriter(fn);

            file.WriteLine(mErrCompleto);
            file.WriteLine("");
            if (TextBox2.Text.Trim().Length > 0)
            {
                file.WriteLine(TextBox2.Text);
                file.WriteLine("");
            }
            //Print #FF, sRep
            file.WriteLine(static_ErrLog_Separator);
            file.Flush();
            file.Close();

        }

    }
}

